src/js/content/content.js

```diff
@@ -17,11 +17,33 @@
 import { findFragmentsInContent } from '../helpers/nodes/nodesHelper';
 import { status } from '../helpers/constants';
 
-window.onload = () => {
-  chrome.runtime.sendMessage({ fragmentsData: findFragmentsInContent() }, (response) => {
-    if (response.status === status.succes) {
-      // eslint-disable-next-line no-console
-      console.log(response.msg);
-    }
-  });
+
+export const getData = async (contentType) => {
+  if (contentType === 'application/json') {
+    const url = window.location.href;
+    const fetchedData = await fetch(url)
```

Comment:
This is not a valid solution. We can not call the url twice - every request can have different outputs. From the other hand, `document.body` content can be modified by other extensions / Chrome. I think we need to rethink the way we get the response body. See more details here: https://medium.com/better-programming/chrome-extension-intercepting-and-reading-the-body-of-http-requests-dd9ebdf2348b